[blog] MUI X v7 beta announcement blogpost (#40784)
mui/material-ui: docs/pages/blog/mui-x-v7-beta.md

  - [Smaller bundle size](#smaller-bundle-size)
- [Tree View](#tree-view)
  - [RichTreeView](#richtreeview-new-component)
- [Charts](#charts)
  - [Reference Line](#reference-line)
- [Date Pickers](#date-pickers)
  - [Support for date-fns v3](#support-for-date-fns-v3)
  - [Date Time Range Picker](#date-time-range-picker)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
- [What's next](#whats-next)
- [Migration guide](#migration-guide)
- [How to get involved](#how-to-get-involved)

## Data Grid

We have introduced critical breaking changes designed not only to enhance the developer experience but also to bring the user experience to a new level.

### Sticky headers

The column headers and pinning implementations have been refactored for an improved user experience.
You'll notice a more responsive Data Grid with smoother horizontal scrolling and better screen reader support.
The video below showcases some of these improvements.

<video preload="metadata" autoplay muted loop playsinline width="690" height="417" controls>
  <source src="/static/blog/mui-x-v7-beta/sticky-headers.mp4" type="video/mp4">
</video>

:::warning

> The new experience is far better, but the solution has a limitation: **column headers can no longer be transparent** because content scrolls behind them.

Reviewer: To be precise, the vertical scroll is worse

```suggestion
The new horizontal scroll experience is far better, but the solution has a limitation: **column headers can no longer be transparent** because content scrolls behind them.
```

> If you're using our theming system, the DataGrid will adapt to your application seamlessly.

Reviewer: ```suggestion
If you're using our theming system, the Data Grid will adapt to your application seamlessly.
```

Without the theming system, you may need to manually set an appropriate background color to maintain your design.
:::

### Improved columns panel design

The column management panel now features a sleeker, checkbox-based design, replacing the previous toggle button style. Additionally, we've extracted the internal component to facilitate the introduction of the upcoming pivoting UI.

<img width="310" alt="new column management panel design" src="/static/blog/mui-x-v7-beta/column-management-panel.png">

### Date Object support in filter model

The `filterModel` now supports `Date` objects for `date` and `dateTime` column types, providing a more intuitive and efficient filtering experience.

While string values remain compatible for these types, any updates to the `filterModel` made through the UI (such as via the filter panel) will now automatically use `Date` objects, ensuring consistency and ease of data handling.

### Smaller bundle size

> The introduction of a separate entry point for locales has significantly optimized our bundle size.

Reviewer: This is misleading otherwise, the reality is that the bundle size has **increased** for the use cases that matter.

```suggestion
The introduction of a separate entry point for locales has significantly optimized the bundle size of the barrel index in development mode.
```

> This change led to a reduction of approximately 22% – shrinking the bundle size from 114.2kB to 88.5kB for `@mui/x-data-grid`.

Reviewer: Only trust what you can verify:

```suggestion

For example with `@mui/x-data-grid`, this change led to a reduction of approximately 22% – shrinking the bundle size from [114.2kB](https://bundlephobia.com/package/@mui/x-data-grid@6.19.2) to [88.5kB](https://bundlephobia.com/package/@mui/x-data-grid@7.0.0-beta.0).
```

> <img width="694" alt="Bundle size change" src="/static/blog/mui-x-v7-beta/new-bundle-size.png">

Reviewer: Makes it easier to read.

```suggestion
<img width="694" alt="Bundle size change" src="/static/blog/mui-x-v7-beta/new-bundle-size.png" style="padding-left: 20px; padding-right: 5px;">
```

You can find more details on its [documentation](https://next.mui.com/x/react-charts/axis/#reference-line).

## Date and Time Pickers

The focus has been mostly on developing the highly anticipated new components, DateTimeRangePicker and TimeRangePicker, alongside a significant effort to enhance accessibility in the fields, ensuring that our components are as inclusive and user-friendly as possible.

While our journey towards these new additions continues, our commitment to improving existing features remains a top priority.
Accordingly, we consistently update existing components to align seamlessly with the latest versions of supported date libraries.

### Date Time Range Picker [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')

Support new use cases with this most anticipated component.
The latest addition to the Date and Time Pickers suite is the `<DateTimeRangePicker />`, a component designed to enrich the user experience with advanced date and time selection.
This intuitive picker simplifies the process of selecting date and time ranges, making it ideal for applications that require detailed scheduling or period selection.

> <video preload="metadata" autoplay muted loop playsinline width="100%" height="510">

Reviewer: To respect the scale and not be blurry

```suggestion
<video preload="metadata" autoplay muted loop playsinline width="600" height="510">
```

date: 2024-01-29T00:00:00.000Z
authors: ['josefreitas']
tags: ['MUI X', 'News']
card: true
---

<div style="max-width: 692px; width: 100%; height: 230px; overflow: hidden; margin-bottom: 16px;">
  <a href="https://github.com/mui/mui-x/releases/tag/v7.0.0-beta.0">
    <img src="/static/blog/mui-x-v7-beta/intro.png" alt="MUI X v7 beta release" width="1200" height="400" style="width: 100%; height: 100%; object-fit: cover; object-position: center;" />
  </a>
</div>

It's the end of January, and we're ready to start rolling out some of the plans for 2024!
We have new components, exciting features, and a lot of improvements for both developers and end-users.

Starting [now](https://github.com/mui/mui-x/releases/tag/v7.0.0-beta.0), MUI X v7 is in beta.
Most breaking changes are in place, and we shift our focus towards refining and expanding over the new version.

## Table of contents

- [Data Grid](#data-grid)
  - [Sticky headers](#sticky-headers)
  - [Server-side data source](#server-side-data-source)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
  - [Date Object Support in Filter Model](#date-object-support-in-filter-model)

>   - [New stable features](#new-stable-features)

Reviewer 2: Optional: Although not substantial, but [this](https://github.com/mui/mui-x/pull/11770) visual update to the columns panel could be a small possible highlight.
```suggestion
  - [New stable features](#new-stable-features)
  - [New columns panel design](#new-columns-panel)
```

- [Tree View](#tree-view)
  - [RichTreeView](#richtreeview-new-component)
- [Charts](#charts)
  - [Reference Line](#reference-line)
- [Date Pickers](#date-pickers)
  - [Support for date-fns v3](#support-for-date-fns-v3)
  - [Date Time Range Picker](#date-time-range-picker)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
- [What's next](#whats-next)
- [Migration guide](#migration-guide)
- [How to get involved](#how-to-get-involved)

## Data Grid

We have introduced critical breaking changes designed not only to enhance the developer experience but also to bring the user experience to a new level.

### Sticky headers

The column headers and pinning implementations have been refactored for an improved user experience.

> You'll notice a more responsive Data Grid with smoother vertical scrolling and better screen reader support.

Reviewer 3: ```suggestion
You'll notice a more responsive Data Grid with smoother horizontal scrolling and better screen reader support.
```

The video below showcases some of these improvements.

<video preload="metadata" style="margin-bottom: 10px;" autoplay muted loop playsinline width="690" height="417" controls>
  <source src="/static/blog/mui-x-v7-beta/sticky-headers.mp4" type="video/mp4">
</video>

:::warning
The new experience is far better, but the solution has a limitation: **column headers can no longer be transparent** because content scrolls behind them.

If you're using our theming system, the DataGrid will adapt to your application seamlessly.

Without the theming system, you may need to manually set an appropriate background color to maintain your design.
:::

### Server-side data source

Integration with the server side has been a huge pain point for our community; we aim to enhance developer experience and productivity by introducing the `DataSource` interface.

> This new interface establishes more efficient data management and seamless interactions, allowing developers to build more responsive and robust applications that connect UI elements to backend services with less complexity.

Reviewer 3: We should make it clear that this feature is not available yet, otherwise it could be misleading

It's initially designed for the Data Grid, but we plan that the DataSource will serve data to multiple components in the future.

```jsx
const myDataSource: DataSource = {
  getRows: async (params: GetRowsParams): GetRowsResponse => {
    // fetch data from server
    const response = await fetch('https://my-api.com/data', {
      method: 'GET',
      body: JSON.stringify(params),
    });
    const data = await response.json();
    // return the data and the total number of rows
    return {
      rows: data.rows,
      rowCount: data.totalCount,
    };
  },
};
```

```jsx
<DataGridPro dataSource={myDataSource} columns={columns} />
```

Learn more on [the server side integration page](https://next.mui.com/x/react-data-grid/server-side-data/#data-source).

> 

Reviewer 3: ```suggestion

### Smaller bundle size

The locales now have a separate entry point. This reduced the bundle size by ~22% (from 114.2kB to 88.5kB for `@mui/x-data-grid`):

<img width="694" alt="Bundle size change" src="https://github.com/mui/material-ui/assets/13808724/11bfbb38-6805-4e54-99a2-e48d544e97c8">

```

### Date Object support in filter model

The `filterModel` now supports `Date` objects for `date` and `dateTime` column types, providing a more intuitive and efficient filtering experience.

While string values remain compatible for these types, any updates to the `filterModel` made through the UI (such as via the filter panel) will now automatically use `Date` objects, ensuring consistency and ease of data handling.

### New stable features

During major version updates, MUI X introduces new features under the `experimentalFeatures` flag or using the `unstable_` prefix.
This approach serves as a failsafe in case there's a need to change the API based on user feedback.

We're excited to announce that the following features have been promoted to stable:

> - [Column groups](https://next.mui.com/x/react-data-grid/column-groups/)

Reviewer 2: ```suggestion
- [Column groups](https://next.mui.com/x/react-data-grid/column-groups/)
- [Header filters](https://next.mui.com/x/react-data-grid/filtering/header-filters/) [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
```

- [Cell selection](https://next.mui.com/x/react-data-grid/cell-selection/) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')
- [Clipboard paste](https://next.mui.com/x/react-data-grid/clipboard/#clipboard-paste) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

> 

Reviewer 2: ```suggestion

## New columns panel design

The columns management panel has been redesigned to have a more sleek checkbox-based design as compared to the previous toggle button-based design. The internal component has also been extracted to pave the way for the new pivoting UI.

  <img width="310" alt="image" src="https://github.com/mui/mui-x/assets/12609561/a79dac8b-d54d-4e69-a63a-ef78f3993f37">

```

## Tree View

Following the promotion of the component from the labs to MUI X, our primary focus has been on improving its developer experience.
This includes efforts to clarify documentation and improve key examples, making them more informative and user-friendly.

A particularly significant initiative is the time-saving alternative to the traditional JSX architecture, the RichTreeView.

### RichTreeView

We've split the Tree View into two distinct components.

The first is the `SimpleTreeView`, which retains the classic and familiar JSX approach and is still the one behind the original `<TreeView />` tag.

> The second is the `RichTreeView` , a new variant designed to streamline the development process, and with which we aim to support several common use cases out-of-the-box.

Reviewer 4: ```suggestion
The second is the `RichTreeView`, a new variant designed to streamline the development process, and with which we aim to support several common use cases out-of-the-box.
```

> Similar to the DataGrid, it requires only a dataset and possibly a few properties to render; it does all the heavy lifting for you, handling most of the complexities internally.

Reviewer 5: ```suggestion
Similar to the Data Grid, it requires only a dataset and possibly a few properties to render; it does all the heavy lifting for you, handling most of the complexities internally.
```

Or 
```suggestion
Similar to the `DataGrid`, it requires only a dataset and possibly a few properties to render; it does all the heavy lifting for you, handling most of the complexities internally.
```

```jsx
const myDataSet = [
  {
    id: 'node-1',
    label: 'Node 1',
    children: [
      { id: 'node-1-1', label: 'Node 1.1' },
      { id: 'node-1-2', label: 'Node 1.2' },
    ],
  },
  {
    id: 'node-2',
    label: 'Node 2',
  },
];
```

```jsx
<RichTreeView
  items={myDataSet}
  defaultCollapseIcon={<ExpandMoreIcon />}
  defaultExpandIcon={<ChevronRightIcon />}
/>
```

Check out the [new component](https://next.mui.com/x/react-tree-view/rich-tree-view/items/)!

## Charts

Since the initial stable release of MUI X Chart a few months ago, we have been diligently listening to your feedback and focusing on refining the foundational charts to ensure they effectively cater to your diverse use cases.

> This includes bug fixes, doc improvements, and support for more complex scenarios.

Reviewer 4: ```suggestion
This includes bug fixes, documentation improvements, and support for more complex scenarios.
```

### Reference line

> The `<ChartsReferenceLine />` component enhances data visualization, providing users with a clear reference to better understand and analyze key data points.

Reviewer 4: Given [the suggestions](https://www.notion.so/mui-org/Style-conventions-for-MUI-components-and-products-dff7d2d7d1ba4a4abd47f48cf345b800?pvs=4#8f86b8e3c0fb49e1b4df3276efb3848e) from @samuelsycamore this would probably even be just bare "ChartsReferenceLine" word, but I'm sure if it's for the better.
IMHO, having some sort of emphasis on the component names makes sense.
WDYT? 🤔 
```suggestion
The `ChartsReferenceLine` component enhances data visualization, providing users with a clear reference to better understand and analyze key data points.
```

<img alt="Charts reference line" src="/static/blog/mui-x-v7-beta/charts-reference-line.png" width="1200" height="840" loading="lazy" style="margin-bottom: 16px;" />

You can find more details on its [documentation](https://next.mui.com/x/react-charts/axis/#reference-line).

> ## Date Pickers

Reviewer 5: ```suggestion
## Date and Time Pickers
```

The focus has been mostly on developing the highly anticipated new components, DateTimeRangePicker and TimeRangePicker, alongside a significant effort to enhance accessibility in the fields, ensuring that our components are as inclusive and user-friendly as possible.

While our journey towards these new additions continues, our commitment to improving existing features remains a top priority.
Accordingly, we consistently update existing components to align seamlessly with the latest versions of supported date libraries.

### Date Time Range Picker [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')

Support new use cases with this most anticipated component.

> The latest addition to the Date Pickers suite is the `<DateTimeRangePicker />`, a component designed to enrich the user experience with advanced date and time selection.

Reviewer 5: ```suggestion
The latest addition to the Date and Time Pickers suite is the `<DateTimeRangePicker />`, a component designed to enrich the user experience with advanced date and time selection.
```

Reviewer 5: Or an alternative

```suggestion
The latest addition to this product line is the `<DateTimeRangePicker />`, a component designed to enrich the user experience with advanced date and time selection.
```

This intuitive picker simplifies the process of selecting date and time ranges, making it ideal for applications that require detailed scheduling or period selection.

<video preload="metadata" style="margin-bottom: 10px;" autoplay muted loop playsinline width="600" height="508">

>   <source src="/static/blog/mui-x-v7-beta/date-time-range-picker.mp4" type="video/mp4">

Reviewer 4: Have you considered using my generated GIF or recording the process on Firefox to have a proper section scrolling behavior? 🤔

</video>

Learn how to use and customize the [new component](https://next.mui.com/x/react-date-pickers/date-time-range-picker/) now!

### Support for date-fns v3

> The Pickers now support both versions of v2 and v3 of date-fns and you can select your desired version from one of the two import lines:

Reviewer 5: ```suggestion
The Date and Time Pickers now support both versions of v2 and v3 of date-fns and you can select your desired version from one of the two import lines:
```

```jsx
// date-fns v2.x
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
```

```jsx
// date-fns v3.x
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFnsV3';
```

## What's next

As we approach the stable release of v7, our roadmap is well-defined, focusing on delivering key enhancements and features either just before or within a couple of weeks following the stable version's launch.

> **Data Grid**

Reviewer 5: Is there a benefit to do bold text instead of an h3 here?

Author: My thought was to avoid any issue with the anchoring links since each component has two anchors with the same name, one for what's new and one for what's next.

- [Pivoting](https://github.com/mui/mui-x/issues/214) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

> - [Improved column management panel](https://github.com/mui/mui-x/issues/5700)

Reviewer 2: ```suggestion
- [Improved column management panel with support for pivoting](https://github.com/mui/mui-x/issues/5700)
- [Data source implementation for tree-data](https://next.mui.com/x/react-data-grid/server-side-data/tree-data/)
```

**Tree View**

- [Drag and drop support for RichTreeView](https://github.com/mui/mui-x/issues/9686)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
- [Checkbox selection for Simple- and RichTreeView](https://github.com/mui/mui-x/issues/214)

**Charts**

- [Click events handling](https://github.com/mui/mui-x/issues/10005)

> **Date Pickers**

Reviewer 5: ```suggestion
**Date and Time Pickers**
```

- [TimeRangePicker](https://github.com/mui/mui-x/issues/4460)
- Field with improved accessibility

## Migration guide

We fully understand that updating a dependency library can sometimes be daunting.
To smooth the transition to the new version, we have documented all breaking changes in our migration guides:

- [Data Grid](https://next.mui.com/x/migration/migration-data-grid-v6/)

> - [Date Pickers](https://next.mui.com/x/migration/migration-pickers-v6/)

Reviewer 5: ```suggestion
- [Date and Time Pickers](https://next.mui.com/x/migration/migration-pickers-v6/)
```

---
title: MUI X v7 is now in beta
description: Check out what's new and what's next for v7 stable.
date: 2024-01-31T00:00:00.000Z
authors: ['josefreitas']
tags: ['MUI X', 'News']
card: true
---

<div style="max-width: 692px; width: 100%; height: 230px; overflow: hidden; margin-bottom: 16px;">
  <a href="https://github.com/mui/mui-x/releases/tag/v7.0.0-beta.0">
    <img src="/static/blog/mui-x-v7-beta/intro.png" alt="MUI X v7 beta release" width="1200" height="400" style="width: 100%; height: 100%; object-fit: cover; object-position: center;" />
  </a>
</div>

It's the end of January, and we're ready to start rolling out some of the plans for 2024!
We have new components, exciting features, and a lot of improvements for both developers and end-users.

Starting [now](https://github.com/mui/mui-x/releases/tag/v7.0.0-beta.0), MUI X v7 is in beta.
Most breaking changes are in place, and we shift our focus towards refining and expanding over the new version.

## Table of contents

> - [What's New](#whats-new)

Reviewer: The heading structure is optimized for one h1, don't think we need this one, it's a blog, it's implicit.

```suggestion
```

  - [Data Grid](#data-grid)
    - [Sticky Headers](#sticky-headers)

>     - [Server-Side Data Source](#server-side-data-source)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')

Reviewer: Sentence case

```suggestion
    - [Server-side data source](#server-side-data-source)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
```

    - [Date Object Support in Filter Model](#date-object-support-in-filter-model)
    - [New Stable Features](#new-stable-features)
  - [TreeView](#treeview)
    - [RichTreeView](#richtreeview-new-component)
  - [Charts](#charts)
    - [Reference Line](#reference-line)
  - [Date Pickers](#date-pickers)

>     - [Date Time Range Picker](#date-time-range-picker)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
>     - [Support for date-fns v3](#support-for-date-fns-v3)

Reviewer: How about we revert the order

```suggestion
    - [Support for date-fns v3](#support-for-date-fns-v3)
    - [Date Time Range Picker](#date-time-range-picker)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
```

> - [What's Next](#whats-next)
> - [Migration Guide](#migration-guide)
> - [How to Get Involved](#how-to-get-involved)

Reviewer: Sentence case

```suggestion
- [What's next](#whats-next)
- [Migration guide](#migration-guide)
- [How to get involved](#how-to-get-involved)
```

> # What's new
> 

Reviewer: The heading structure is optimized for one h1, don't think we need this one, it's a blog, it's implicit.

```suggestion
```

## Data Grid

We have introduced critical breaking changes designed not only to enhance the developer experience but also to bring the user experience to a new level.

### Sticky Headers

The column headers and pinning implementations have been refactored for a significantly better user experience.
You'll notice a more responsive Data Grid with smoother scrolling and better screen reader support.
The video below showcases some of these improvements.

> <video preload="metadata" style="margin-bottom: 10px;" autoplay muted loop playsinline width="690" height="417">

Reviewer: Maybe to add the controls, I wanted to play with the video at one point but couldn't. It's longer than 10s so could be better:

```suggestion
<video preload="metadata" style="margin-bottom: 10px;" autoplay muted loop playsinline width="690" height="417" controls>
```

  <source src="/static/blog/mui-x-v7-beta/sticky-headers.mp4" type="video/mp4">
</video>

:::warning
The new experience is far better, but the solution has a limitation: **column headers can no longer be transparent** because content scrolls behind them.

If you're using our theming system, the DataGrid will adapt to your application seamlessly.

Without the theming system, you may need to manually set an appropriate background color to maintain your design.
:::

> ### Server-Side Data Source

Reviewer: Sentence case

```suggestion
### Server-side data source
```

    // fetch data from server
    const response = await fetch('https://my-api.com/data', {
      method: 'GET',
      body: JSON.stringify(params),
    });
    const data = await response.json();
    // return the data and the total number of rows
    return {
      rows: data.rows,
      rowCount: data.totalCount,
    };
  },
};
```

```jsx
<DataGridPro dataSource={myDataSource} columns={columns} />
```

Learn more on [the server side integration page](https://next.mui.com/x/react-data-grid/server-side-data/#data-source).

### Date Object support in filter model

The `filterModel` now supports `Date` objects for `date` and `dateTime` column types, providing a more intuitive and efficient filtering experience.

While string values remain compatible for these types, any updates to the `filterModel` made through the UI (such as via the filter panel) will now automatically use `Date` objects, ensuring consistency and ease of data handling.

### New stable features

During major versions, MUI X releases new features under the `experimental` flag as a failsafe in case there's a need to change the API based on user feedback.

> We're now excited to announce that the following features have been promoted to stable and don't require the use of the flag anymore.

Reviewer: ```suggestion
We're now excited to announce that the following features have been promoted to stable and don't require the use of the `unstable_` flag anymore:
```

> - [Cell selection](https://next.mui.com/x/react-data-grid/cell-selection/) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')
> - [Clipboard pasting](https://next.mui.com/x/react-data-grid/clipboard/#clipboard-paste) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')
> - [Column grouping](https://next.mui.com/x/react-data-grid/column-groups/)

Reviewer: Seems clearer by plan:

```suggestion
- [Column groups](https://next.mui.com/x/react-data-grid/column-groups/)
- [Cell selection](https://next.mui.com/x/react-data-grid/cell-selection/) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')
- [Clipboard paste](https://next.mui.com/x/react-data-grid/clipboard/#clipboard-paste) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')
```

> ## TreeView

Reviewer: Component name

```suggestion
## Tree View
```

Following the promotion of the component from the labs to MUI X, our primary focus has been on improving its developer experience.
This includes efforts to clarify documentation and improve key examples, making them more informative and user-friendly.

A particularly significant initiative is the time-saving alternative to the traditional JSX architecture, the RichTreeView.

### RichTreeView

> We've evolved the TreeView into two distinct components.

Reviewer 6: 'evolved' sounds a bit strange here 🤔 Maybe replacing it with 'restructured' or 'separated' could work?

The first is the `SimpleTreeView`, which retains the classic and familiar JSX approach and is still the one behind the original `<TreeView />` tag.

The second is the `RichTreeView` , a new variant designed to streamline the development process, and with which we aim to support several common use cases out-of-the-box.

Similar to the DataGrid, it requires only a dataset and possibly a few properties to render; it does all the heavy lifting for you, handling most of the complexities internally.

```jsx

> const myDataSet = [
>     {
>         id: 'node-1',
>         label: 'Node 1',
>         children: [
>             { id: 'node-1-1', label: Node 1.1' },
>             { id: 'node-1-2', label: Node 1.2' },
>         ],
>     },
>         {
>         id: 'node-2',
>         label: 'Node 2',
>     },
> ];

Reviewer: Missing quote and indentation:

```suggestion
const myDataSet = [
  {
    id: 'node-1',
    label: 'Node 1',
    children: [
      { id: 'node-1-1', label: 'Node 1.1' },
      { id: 'node-1-2', label: 'Node 1.2' },
    ],
  },
  {
    id: 'node-2',
    label: 'Node 2',
  },
];
```

```

```jsx
<RichTreeView
  items={myDataSet}
  defaultCollapseIcon={<ExpandMoreIcon />}

>   defaultExpandIcon={<ChevronRightIcon />}

Author: I think it's interesting to show that you're passing items + props.

>   defaultExpandIcon={<ChevronRightIcon />}

Reviewer 5: Maybe another prop then
Those ones were mandatory in v6 so keeping them might make people think that they still are
How about using `multiSelect` for example, or `onNodeExpansionToggle`

/>
```

> Checkout the [new component](https://next.mui.com/x/react-tree-view/rich-tree-view/items/)!

Reviewer 6: ```suggestion
Check out the [new component](https://next.mui.com/x/react-tree-view/rich-tree-view/items/)!
```

## Charts

> Since the initial stable release of the charts components a few months ago, we have been diligently listening to your feedback and focusing on refining the foundational charts to ensure they effectively cater to your diverse use cases.

Reviewer 6: Super nitpicky comment: both sentences end with 'use cases' and sounds a bit repetitive to me 🙈 Maybe we can replace one with 'scenarios'

> Since the initial stable release of the charts components a few months ago, we have been diligently listening to your feedback and focusing on refining the foundational charts to ensure they effectively cater to your diverse use cases.

Reviewer: I think it's best to treat this as its own mini-open source project

```suggestion
Since the initial stable release of MUI X Chart a few months ago, we have been diligently listening to your feedback and focusing on refining the foundational charts to ensure they effectively cater to your diverse use cases.
```

While our journey towards these new additions continues, our commitment to improving existing features remains a top priority.
Accordingly, we consistently update existing components to align seamlessly with the latest versions of supported date libraries.

### Date Time Range Picker [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')

Support new use cases with this most anticipated component.
The latest addition to the Date Pickers suite is the `<DateTimeRangePicker />`, a component designed to enrich the user experience with advanced date and time selection.
This intuitive picker simplifies the process of selecting date and time ranges, making it ideal for applications that require detailed scheduling or period selection.

<video preload="metadata" style="margin-bottom: 10px;" autoplay muted loop playsinline width="600" height="508">
  <source src="/static/blog/mui-x-v7-beta/date-time-range-picker.mp4" type="video/mp4">
</video>

Learn how to use and customize the [new component](https://next.mui.com/x/react-date-pickers/date-time-range-picker/) now!

> ### Support to date-fns v3

Reviewer: ```suggestion
### Support for date-fns v3
```

The Pickers now support both versions of v2 and v3 of date-fns and you can select your desired version from one of the two import lines:

```jsx
// date-fns v2.x
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
```

```jsx
// date-fns v3.x
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFnsV3';
```

> # What's next

Reviewer: Can't have multiple h1

```suggestion
## What's next
```

As we approach the stable release of v7, our roadmap is well-defined, focusing on delivering key enhancements and features either just before or within a couple of weeks following the stable version's launch.

**Data Grid**

- [Pivoting](https://github.com/mui/mui-x/issues/214) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')
- [Improved column management panel](https://github.com/mui/mui-x/issues/5700)

**TreeView**

- [Drag and drop support for RichTreeView](https://github.com/mui/mui-x/issues/9686)[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
- [Checkbox selection for Simple- and RichTreeView](https://github.com/mui/mui-x/issues/214)

**Charts**

- [Click events handling](https://github.com/mui/mui-x/issues/10005)

**Date pickers**

- [TimeRangePicker](https://github.com/mui/mui-x/issues/4460)
- Field with improved accessibility

> # Migration guide

Reviewer: Can't have multiple h1

```suggestion
## Migration guide
```

We fully understand that updating a dependency library can sometimes be daunting.
To smooth the transition to the new version, we have documented all breaking changes in our migration guides:

- [Data Grid](https://next.mui.com/x/migration/migration-data-grid-v6/)

> - [Date pickers](https://next.mui.com/x/migration/migration-date-pickers-v6/)

Reviewer: ```suggestion
- [Date Pickers](https://next.mui.com/x/migration/migration-pickers-v6/)
```

- [Charts](https://next.mui.com/x/migration/migration-charts-v6/)

> - [TreeView](https://next.mui.com/x/migration/migration-tree-view-v6/)

Reviewer: ```suggestion
- [Tree View](https://next.mui.com/x/migration/migration-tree-view-v6/)
```

These component-specific guides are reference pages designed to assist you if you encounter any challenges while updating to the new major version.
It's our way of making the upgrade process more manageable and less burdensome for you.
However, should you need further assistance, please don't hesitate to contact us and send your feedback.

> # How to get involved

Reviewer: Can't have multiple h1

```suggestion
## How to get involved
```

Your feedback has been invaluable in developing MUI X, and we're always happy to hear from you. Please consider sharing your experiences and pain points by:

- [Giving us a user interview](https://forms.gle/vsBv6CLPz9h57xg8A).
- Reporting bugs and suggesting features on our [GitHub repository](https://github.com/mui/mui-x/issues/new/choose).

> We look forward to your input!X

Reviewer: ```suggestion
We look forward to your input!
```
